src/Compilers/CSharp/Test/Semantic/Semantics/RecordStructTests.cs

```diff
@@ -1292,6 +1399,1821 @@ public unsafe record C
+                );
+        }
+
+        [Fact]
+        public void RecordProperties_PropertyInValueTuple_Readonly()
+        {
+            var corlib_cs = @"
+namespace System
+{
+    public class Object { }
+    public class ValueType
+    {
+        public bool X { get; set; }
+    }
+    public class Attribute { }
+    public struct Void { }
+    public struct Boolean { }
+}
+";
+            var corlibRef = CreateEmptyCompilation(corlib_cs).EmitToImageReference();
+
+            var src = @"
+readonly record struct C(bool X)
+{
+    bool M()
+    {
+        return X;
+    }
+}
+";
+            var comp = CreateEmptyCompilation(src, parseOptions: TestOptions.RegularPreview, references: new[] { corlibRef });
+            comp.VerifyEmitDiagnostics(
+                // (2,31): warning CS8907: Parameter 'X' is unread. Did you forget to use it to initialize the property with that name?
+                // readonly record struct C(bool X);
+                Diagnostic(ErrorCode.WRN_UnreadRecordParameter, "X").WithArguments("X").WithLocation(2, 31)
+                );
+
+            Assert.Null(comp.GlobalNamespace.GetTypeMember("C").GetMember("X"));
+            var tree = comp.SyntaxTrees.Single();
+            var model = comp.GetSemanticModel(tree, ignoreAccessibility: false);
+            var x = tree.GetRoot().DescendantNodes().OfType<ReturnStatementSyntax>().Single().Expression;
+            Assert.Equal("System.Boolean System.ValueType.X { get; set; }", model.GetSymbolInfo(x!).Symbol.ToTestDisplayString());
+        }
+
+        [Fact]
+        public void StaticCtor()
+        {
+            var src = @"
+System.Console.Write(""static ctor"");
+
+record struct R(int x);
+";
+
+            var comp = CreateCompilation(src);
+            comp.VerifyEmitDiagnostics();
+            CompileAndVerify(comp, expectedOutput: "static ctor");
+        }
+
+        [Fact]
+        public void StaticCtor_ParameterlessPrimaryCtor()
+        {
+            var src = @"
+record struct R()
+{
+    static R() { }
+}
+";
+
+            var comp = CreateCompilation(src);
+            comp.VerifyEmitDiagnostics();
+        }
+
+        [Fact]
+        public void StaticCtor_CopyCtor()
+        {
+            var src = @"
+record struct R()
+{
+    static R(R r) { }
+}
+";
+
+            var comp = CreateCompilation(src);
+            comp.VerifyEmitDiagnostics(
+                // (4,12): error CS0132: 'R.R(R)': a static constructor must be parameterless
+                //     static R(R r) { }
+                Diagnostic(ErrorCode.ERR_StaticConstParam, "R").WithArguments("R.R(R)").WithLocation(4, 12)
+                );
+        }
+
+        [Fact]
+        public void InterfaceImplementation_NotReadonly()
+        {
+            var source = @"
+I r = new R(42);
+r.P2 = 43;
+r.P3 = 44;
+System.Console.Write((r.P1, r.P2, r.P3));
+
+interface I
+{
+    int P1 { get; set; }
```

Comment:
>set [](start = 18, length = 3)

Are we testing scenario with init setter and mutable record?

Reply:
See `PositionalMemberDefaultValue_AndPropertyWithInitializer_CopyingParameter`

---
In reply to: [590823028](https://github.com/dotnet/roslyn/pull/51720#discussion_r590823028) [](ancestors = 590823028)

Reply:
> See PositionalMemberDefaultValue_AndPropertyWithInitializer_CopyingParameter

That doesn't look like what I am looking for. I am looking for scenario that matches this one with the only change where P1 is init-only in the interface

---
In reply to: [590826814](https://github.com/dotnet/roslyn/pull/51720#discussion_r590826814) [](ancestors = 590826814,590823028)